src/oracle/oci.rs

```diff
@@ -0,0 +1,89 @@
+//! Oracle Cloud Infrastructure Singer
+
+use anyhow::{Error, Result};
+use base64::{engine::general_purpose, Engine as _};
+use http::HeaderValue;
+use log::debug;
+use openssl::hash::MessageDigest;
+use openssl::pkey::PKey;
```

Comment:
Can we use `ring` here instead?

Reply:
how about we use https://crates.io/crates/rsa, this is a pure rust signing lib, maybe it could reduce the platform issues.

I have checked the `ring` crate, it combines asm, c and rust, maybe there would also be platform-specific problems.

Reply:
Using rsa is even better!